Lib/ast.py

```diff
@@ -702,11 +703,20 @@ def maybe_newline(self):
         if self._source:
             self.write("\n")
 
-    def fill(self, text=""):
+    def maybe_semicolon(self):
+        """Adds a "; " delimiter if it isn't the start of generated source"""
+        if self._source:
+            self.write("; ")
+
+    def fill(self, text="", allowsemi=True):
```

Comment:
I would suggest allow_semi or newlines=False

Reply:
allow_semi because newlines can still happen

Reply:
Please could we use `allow_semicolon`? Better to use the full name.